src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsCounters.cs

```diff
@@ -10,203 +11,289 @@ internal sealed class WindowsCounters
 {
     public WindowsCounters(IMeterFactory meterFactory, TcpTableInfo tcpTableInfo)
     {
+        const string NetworkStateKey = "system.network.state";
+        const string InstrumentName = "process.network.connections";
+
 #pragma warning disable CA2000 // Dispose objects before losing scope
         // We don't dispose the meter because IMeterFactory handles that
         // An issue on analyzer side: https://github.com/dotnet/roslyn-analyzers/issues/6912
         // Related documentation: https://github.com/dotnet/docs/pull/37170
         var meter = meterFactory.Create("Microsoft.Extensions.Diagnostics.ResourceMonitoring");
 #pragma warning restore CA2000 // Dispose objects before losing scope
 
-        _ = meter.CreateObservableGauge(
-            "ipv4_tcp_connection_closed_count",
+        var tcpTag = new KeyValuePair<string, object?>("network.transport", "tcp");
+
+        // These are covered in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-metrics.md#attributes:
+        var tcpVersionFourTag = new KeyValuePair<string, object?>("network.type", "ipv4");
+        var tcpVersionSixTag = new KeyValuePair<string, object?>("network.type", "ipv6");
+
+        // These metrics are aligned with
+        // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#metric-systemnetworkconnections
+
+        // IPv4:
+        _ = meter.CreateObservableUpDownCounter(
+            InstrumentName,
```

Comment:
Are multiple observable counters with the same name and different tags supported? I haven't used this pattern at all.

Has someone verified this produces expected data in something like the Prometheus exporter scrape exprter?

Reply:
OTel may tolerate it, but it isn't recommended and it will probably generate warnings at minimum. https://opentelemetry.io/docs/specs/otel/metrics/sdk/#duplicate-instrument-registration

The intended pattern is to create a single ObservableUpDownCounter and then use the constructor overload where you provide a callback that returns an IEnumerable<Measurement> instead of a single Measurement. This will let you enumerate multiple tag combinations and values with a single instrument.

Reply:
Will update these to have just one instrument

Reply:
Done